### PR TITLE
Wait for processing thread to finish in LonelyProcessingStepTest

### PR DESCRIPTION
**Pull request by MishaDemianenko**

### CODE REVIEW

**Review by tinwelint — CHANGES_REQUESTED** [review]
I think that changing aspects of the unit in a test is not ideal. I think the test should do what many other step tests do, properly await completion, like so:

```
diff --git a/community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/LonelyProcessingStepTest.java b/community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/LonelyProcessingStepTest.java
index f3f7d6d..65167a0 100644
--- a/community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/LonelyProcessingStepTest.java
+++ b/community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/LonelyProcessingStepTest.java
@@ -36,17 +36,18 @@ public class LonelyProcessingStepTest
     public static SuppressOutput mute = SuppressOutput.suppressAll();

     @Test
-    public void issuePanicBeforeCompletionOnError()
+    public void issuePanicBeforeCompletionOnError() throws InterruptedException
     {
         List<Step<?>> stepsPipeline = new ArrayList<>();
-        BinaryLatch endOfUpstreamLatch = new BinaryLatch();
-        FaultyLonelyProcessingStepTest faultyStep =
-                new FaultyLonelyProcessingStepTest( stepsPipeline, endOfUpstreamLatch );
+        FaultyLonelyProcessingStepTest faultyStep = new FaultyLonelyProcessingStepTest( stepsPipeline );
         stepsPipeline.add( faultyStep );

         faultyStep.receive( 1, null );

-        endOfUpstreamLatch.await();
+        while ( !faultyStep.isCompleted() )
+        {
+            Thread.sleep( 10 );
+        }

         Assert.assertTrue( "On upstream end step should be already on panic in case of exception",
                 faultyStep.isPanicOnEndUpstream() );
@@ -57,14 +58,12 @@ public class LonelyProcessingStepTest

     private class FaultyLonelyProcessingStepTest extends LonelyProcessingStep
     {
-        private final BinaryLatch endOfUpstreamLatch;
         private volatile boolean panicOnEndUpstream;

-        FaultyLonelyProcessingStepTest( List<Step<?>> pipeLine, BinaryLatch endOfUpstreamLatch )
+        FaultyLonelyProcessingStepTest( List<Step<?>> pipeLine )
         {
             super( new StageExecution( "Faulty", null, Configuration.DEFAULT, pipeLine, 0 ),
                     "Faulty", Configuration.DEFAULT );
-            this.endOfUpstreamLatch = endOfUpstreamLatch;
         }

         @Override
@@ -78,10 +77,9 @@ public class LonelyProcessingStepTest
         {
             panicOnEndUpstream = isPanic();
             super.endOfUpstream();
-            endOfUpstreamLatch.release();
         }

-        public boolean isPanicOnEndUpstream()
+        boolean isPanicOnEndUpstream()
         {
             return panicOnEndUpstream;
         }
```

**Comment by MishaDemianenko** [comment]
@tinwelint here we go, updated with suggestion

File: community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/LonelyProcessingStepTest.java [hunk 1]
```diff
@@ -75,15 +86,20 @@ protected void process()
             throw new RuntimeException( "Process exception" );
         }
 
+        @Override
```
Comment (tinwelint) [inline community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/staging/LonelyProcessingStepTest.java]: So you're still changing the implementation of `LonelyProcessingStep` here, is that really necessary?

**Comment by MishaDemianenko** [comment]
@tinwelint now its only waiting that was added

**Review by tinwelint — APPROVED** [review]
Awesome, that should be enough!

**Pull request merged**